kotsadm/pkg/downstream/downstream.go

```diff
@@ -134,3 +136,84 @@ func SetIgnorePreflightPermissionErrors(appID string, sequence int64) error {
 
 	return nil
 }
+
+func GetDownstreamOutput(appID string, clusterID string, sequence int64) (*types.DownstreamOutput, error) {
+	db := persistence.MustGetPGSession()
+	query := `SELECT
+	adv.status,
+	adv.status_info,
+	ado.dryrun_stdout,
+	ado.dryrun_stderr,
+	ado.apply_stdout,
+	ado.apply_stderr
+FROM
+	app_downstream_version adv
+LEFT JOIN
+	app_downstream_output ado
+ON
+	adv.app_id = ado.app_id AND adv.cluster_id = ado.cluster_id AND adv.sequence = ado.downstream_sequence
+WHERE
+	adv.app_id = $1 AND
+	adv.cluster_id = $2 AND
+	adv.sequence = $3`
+	row := db.QueryRow(query, appID, clusterID, sequence)
+
+	var status sql.NullString
+	var statusInfo sql.NullString
+	var dryrunStdout sql.NullString
+	var dryrunStderr sql.NullString
+	var applyStdout sql.NullString
+	var applyStderr sql.NullString
+
+	if err := row.Scan(&status, &statusInfo, &dryrunStdout, &dryrunStderr, &applyStdout, &applyStderr); err != nil {
+		if err == sql.ErrNoRows {
+			return &types.DownstreamOutput{
```

Comment:
`return &types.DownstreamOutput{}` is enough here.

Reply:
ah, you're right, trying to mimic the logic in TS is sometimes a trap :)